src/compas_fab/backends/client.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import print_function
+
+
+class ClientInterface(object):
+    def __init__(self):
+        self.planner = PlannerInterface(self)
+        # self.control = ControlInterface()
+
+    def inverse_kinematics(self, *args, **kwargs):
+        return self.planner.inverse_kinematics(*args, **kwargs)
+
+    def forward_kinematics(self, *args, **kwargs):
+        return self.planner.forward_kinematics(*args, **kwargs)
+
+    def plan_cartesian_motion(self, *args, **kwargs):
+        return self.planner.plan_cartesian_motion(*args, **kwargs)
+
+    def plan_motion(self, *args, **kwargs):
+        return self.planner.plan_motion(*args, **kwargs)
+
+    # ==========================================================================
+    # collision objects and planning scene
+    # ==========================================================================
+
+    def get_planning_scene(self, *args, **kwargs):
+        return self.planner.get_planning_scene(*args, **kwargs)
+
+    def add_collision_mesh(self, collision_mesh):
+        return self.planner.add_collision_mesh(collision_mesh)
+
+    def remove_collision_mesh(self, id):
+        return self.planner.remove_collision_mesh(id)
+
+    def append_collision_mesh(self, collision_mesh):
+        return self.planner.append_collision_mesh(collision_mesh)
+
+    def add_attached_collision_mesh(self, attached_collision_mesh):
+        return self.planner.add_attached_collision_mesh(attached_collision_mesh)
+
+    def remove_attached_collision_mesh(self, id):
+        return self.planner.remove_attached_collision_mesh(id)
+
+#     # ==========================================================================
+#     # executing
+#     # ==========================================================================
+#
+#     def execute_joint_trajectory(self, *args, **kwargs):
+#         self.control.execute_joint_trajectory(*args, **kwargs)
+#
+#     def follow_joint_trajectory(self, *args, **kwargs):
+#         self.control.follow_joint_trajectory(*args, **kwargs)
+#
+#     def get_configuration(self, *args, **kwargs):
+#         self.control.get_configuration(*args, **kwargs)
+#
+#
+# class ControlInterface(object):
+#     def execute_joint_trajectory(self, *args, **kwargs):
+#         raise NotImplementedError('Assigned control does not have this feature.')
+#
+#     def follow_joint_trajectory(self, *args, **kwargs):
+#         raise NotImplementedError('Assigned control does not have this feature.')
+#
+#     def get_configuration(self, *args, **kwargs):
+#         raise NotImplementedError('Assigned control does not have this feature.')
+
+
+class PlannerInterface(object):
+    def __init__(self, client):
+        self.client = client
+
+    def init_planner(self):
+        pass
+
+    def dispose_planner(self):
+        pass
+
+    def validate_response(self, response):
```

Comment:
Not sure if `validate_response` is so general in its current shape, it's more of a MoveIt thing (at least with that signature)